### PR TITLE
Add the Android example app

### PR DESCRIPTION
**Pull request by programmablereya**
This creates a simple Android app and the necessary BUILD files, as well as necessary changes to the WORKSPACE.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,7 +38,16 @@ localhost:8080 and get a response from the server.
 Android application
 -------------------
 
-TODO
+The Android application has one activity, which retrieves the `'requested'` key
+of the JSON object retrieved from the URL http://10.0.0.2:8080/ and displays it
```
Comment (kchodorow) [inline README.md]: http://localhost:8080

File: WORKSPACE [hunk 2]
```diff
@@ -30,7 +30,22 @@ maven_jar(
     artifact = "javax.servlet:servlet-api:2.5",
 )
 
+maven_jar(
```
Comment (kchodorow) [inline WORKSPACE]: Move below javax/servlet/api

File: README.md [hunk 3]
```diff
@@ -38,7 +38,16 @@ localhost:8080 and get a response from the server.
...
+on the screen.
+
+You can build and install this application by connecting an Android emulator or
+device via adb and running:
```
Comment (kchodorow) [inline README.md:53]: Also mention that you need to download the sdk from &lt;url&gt; and change the WORKSPACE file.

File: android/src/main/java/com/google/bazel/example/android/activities/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          if (result == null) {
+            Toast.makeText(
+                    MainActivity.this, getString(R.string.error_sending_request), Toast.LENGTH_LONG)
+                .show();
+            textView.setText("???");
+            return;
+          }
+          textView.setText(result);
+        }
+      }.execute("http://10.0.2.2:8080/boop");
```
Comment (kchodorow) [inline android/src/main/java/com/google/bazel/example/android/activities/MainActivity.java:118]: localhost?

File: android/src/main/java/com/google/bazel/example/android/res/values/strings.xml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">HelloBazel</string>
+</resources>
```
Comment (kchodorow) [inline android/src/main/java/com/google/bazel/example/android/res/values/strings.xml]: eof newline?

File: WORKSPACE [hunk 6]
```diff
@@ -30,7 +30,22 @@ maven_jar(
...
+
+android_sdk_repository(
+  name = "androidsdk",
+  # Set the path to the directory the Android SDK was unzipped into.
+  # path = "/path/to/android-sdk",
+  # Set the API level of the installed SDK Platform.
+  # api_level = 22,
+  # Set the version of the build tools (a directory inside build-tools)
+  # build_tools_version="22.0.1"
+)
```
Comment (kchodorow) [inline WORKSPACE]: eof newline

File: README.md (hunk 1 above)
Comment (programmablereya) [inline README.md]: I've fixed it to match the URL actually used in the app, but localhost is the phone, even if the phone is an emulator. 10.0.2.2 is the host device for an emulator, and thus what is used here.

File: README.md (hunk 3 above)
Comment (programmablereya) [inline README.md:53]: Done.

File: WORKSPACE (hunk 2 above)
Comment (programmablereya) [inline WORKSPACE]: Done.

File: WORKSPACE (hunk 6 above)
Comment (programmablereya) [inline WORKSPACE]: Done.

File: android/src/main/java/com/google/bazel/example/android/activities/MainActivity.java (hunk 4 above)
Comment (programmablereya) [inline android/src/main/java/com/google/bazel/example/android/activities/MainActivity.java:118]: See previous comment, but basically localhost won't work.

File: android/src/main/java/com/google/bazel/example/android/res/values/strings.xml (hunk 5 above)
Comment (programmablereya) [inline android/src/main/java/com/google/bazel/example/android/res/values/strings.xml]: Done.

File: README.md (hunk 1 above)
Comment (kchodorow) [inline README.md]: Ah, got it.

**Pull request merged**